src/isolate/server/_server.py

```diff
@@ -0,0 +1,106 @@
+import secrets
+import threading
+from typing import Dict, Final
+
+from flask import Flask, request
+from marshmallow import ValidationError
+
+from isolate import prepare_environment
+from isolate.backends import BaseEnvironment
+from isolate.backends.context import GLOBAL_CONTEXT
+from isolate.server._models import Environment, Environments, with_schema
+from isolate.server._runs import RunInfo, dehydrated_dual_env_run
+from isolate.server._utils import error, load_token, success
+
+app = Flask(__name__)
+
+
+ENV_STORE: Dict[str, BaseEnvironment] = {}
+RUN_STORE: Dict[str, RunInfo] = {}
+MAX_JOIN_WAIT: Final[float] = 1
+
+
+@app.route("/environments/create", methods=["POST"])
+def create_environment():
+    """Create a new environment from the POST'd definition
+    (as JSON). Returns the token that can be used for running
+    it in the future."""
+    try:
+        data = with_schema(Environment, request.json)
+    except ValidationError as err:
+        return error(message=str(err))
+
+    environment_kind = data["kind"]
+    if environment_kind not in ("conda", "virtualenv"):
+        return error(message=f"Unknown environment kind: {environment_kind}")
+
+    token = secrets.token_urlsafe()
+    ENV_STORE[token] = prepare_environment(
+        environment_kind,
+        **data["configuration"],
+    )
+    return success(token=token)
+
+
+@app.route("/environments/run", methods=["POST"])
+def run_environment():
+    """Run the function (serialized with cloudpickle) from the POST'd data
+    on the environments specified as tokens (in query params). It will return
+    a new token that can be used to check the status of the run periodically
+    through /environment/status endpoint."""
+    try:
+        data = with_schema(Environments, request.args)
+        primary_env = load_token(ENV_STORE, data["primary_token"])
+        secondary_env = load_token(ENV_STORE, data["secondary_token"])
+    except ValidationError as err:
+        return error(message=str(err))
+
+    run_info = RunInfo()
+
+    # Context can allow us to change the serialization backend
+    # and the log processing.
+    run_context = GLOBAL_CONTEXT._replace(
+        _serialization_backend=data["serialization_backend"],
+        _log_handler=run_info.logs.append,
+    )
+    primary_env.set_context(run_context)
+    secondary_env.set_context(run_context)
+
+    # We start running the environment (both the build and the run, actually)
+    # in a separate thread, since we don't want this request to block.
+    run_info.bound_thread = threading.Thread(
+        target=dehydrated_dual_env_run,
+        args=(primary_env, secondary_env, request.data, run_info.done_signal),
+    )
+    run_info.bound_thread.start()
+
+    RUN_STORE[run_info.token] = run_info
+    return success(token=run_info.token)
+
+
+@app.route("/environments/status/<token>", methods=["GET"])
```

Comment:
```suggestion
@app.route("/environments/runs/<token>/status", methods=["GET"])
```

Current route seems to be about `environements`, it is actually about an env run